### DISCUSSION TITLE
Smart Filter nested AND and ORs

### DISCUSSION THREAD

**Discussion by rgecko** [Kareadita/Kavita#2836] [Ideas]
### Idea Description

Would be good to be able to nest "AND" and "ORs" when building smart filters.  
when filtering on an open search option like Series Name, being able to include REGEX or a GUI option to create "OR" type statements.

ComicRack used to do this well for smart filters, where you could create at least nested "ANY" or "ALL" filters under each other so you could filter "ALL" on a publisher or set of publishers as a set of "ANDs", then have an open filter to ANY that allowed for all your "OR" options.

### Idea Category

Feature Enhancement

### Duration of Using Kavita

1 year

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
I'll just let anyone know that when I first implemented the filter revamp I did it this way. But the UI was too conplex to code that I simplified it. So if people really want it, then a PR is likely needed.
